pkg/pool-utils/test/ProtocolAUMFees.test.ts

```diff
@@ -0,0 +1,83 @@
+import { BigNumber, Contract } from 'ethers';
+import { expect } from 'chai';
+import { random } from 'lodash';
+
+import { BigNumberish, bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { DAY } from '@balancer-labs/v2-helpers/src/time';
+
+describe('ProtocolAUMFees', function () {
+  let lib: Contract;
+
+  sharedBeforeEach(async function () {
+    lib = await deploy('MockProtocolAUMFees');
+  });
+
+  function expectedAUMFees(
+    totalSupply: BigNumberish,
+    aumFeePercentage: BigNumberish,
+    timeElapsed: BigNumberish
+  ): BigNumber {
+    return bn(totalSupply)
+      .mul(timeElapsed)
+      .div(365 * DAY)
+      .mul(aumFeePercentage)
+      .div(fp(1).sub(aumFeePercentage));
+  }
+
+  context('when no time has passed since last collection', () => {
+    const lastCollectionTime = random(1000, 10000);
+    const currentTime = lastCollectionTime - 1;
+
+    it('returns zero', async () => {
+      const totalSupply = fp(random(1, 100));
+      const aumFeePercentage = fp(random(0, 0.9));
+
+      expect(await lib.getAumFeesBptAmount(totalSupply, currentTime, lastCollectionTime, aumFeePercentage)).to.be.eq(
+        fp(0)
+      );
+    });
+  });
+
+  context('when time has passed since last collection', () => {
+    const lastCollectionTime = random(1000, 10000);
+    const currentTime = lastCollectionTime + random(1, 365) * DAY;
+
+    context('when AUM fee percentage is zero', () => {
+      const aumFeePercentage = fp(0);
+
+      it('returns zero', async () => {
+        const totalSupply = fp(1);
+
+        expect(await lib.getAumFeesBptAmount(totalSupply, currentTime, lastCollectionTime, aumFeePercentage)).to.be.eq(
+          fp(0)
+        );
+      });
+    });
+
+    context('when AUM fee percentage is non-zero', () => {
+      const aumFeePercentage = fp(random(0, 0.9));
+
+      context('when total supply is zero', () => {
+        const totalSupply = fp(0);
+
+        it('returns zero', async () => {
+          expect(
+            await lib.getAumFeesBptAmount(totalSupply, currentTime, lastCollectionTime, aumFeePercentage)
+          ).to.be.eq(fp(0));
+        });
+      });
+
+      context('when total supply is nonzero', () => {
+        const totalSupply = fp(random(1, 100));
+
+        it('returns the expected amount', async () => {
+          const expectedBptAmount = expectedAUMFees(totalSupply, aumFeePercentage, currentTime - lastCollectionTime);
+          expect(
+            await lib.getAumFeesBptAmount(totalSupply, currentTime, lastCollectionTime, aumFeePercentage)
+          ).to.be.eq(expectedBptAmount);
+        });
```

Comment:
I think it'd be a good practice to have simple numerical examples in the source code to make sure the reader understands what we're trying to do, and to then have a very dumb test that uses those constants to check the result is correct. This would be low effort, would improve readability, and would decrease the burden of the helper functions (in this case `expectedAUMFees`) being correct.

An an example, we could here describe a scenario where total supply is 1000, 7 days have passed, and the fee is 5%. 
7 days is ~1.92% of the year, and so we need to mint xyz. 

Writing this actually made me thing that we're perhaps getting APR and APY mixed up - are you sure we should compute the anual percentage and then divide instead of computing the e.g. daily fee % and then applying that? Regardless of which one is the right one, this seems like good indication of the need to have these examples I suggest.